subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc

```diff
@@ -702,6 +702,55 @@ tasks.create("myTask") {
 // === `Property`, `Provider` and `NamedDomainObjectProvider`
 
 
+[[kotdsl:assignment]]
+== Kotlin Property assignment
+
+Gradle's Kotlin DSL supports Property assignment as an experimental feature that can be enabled by adding `systemProp.org.gradle.unsafe.kotlin.assignment=true` to the `gradle.properties` file.
+Property assignment reduces the verbosity for Kotlin DSL when <<lazy_configuration#lazy_properties,lazy properties>> are used.
+Property assignment works for properties that are _final_ and have type _Property_ or _ConfigurableFileCollection_.
+Property assignment should be preferred over custom property setters since it makes Kotlin DSL code style consistent.
+
+.Kotlin Property assignment
+====
+include::sample[dir="snippets/kotlinDsl/assignment/kotlin",files="build.gradle.kts[tags=assignment]"]
+====
+<1> Set value old way, with `.set()` method (still supported)
+<2> Set value with Property assignment with `=` operator
+<3> Property assignment also supports lazy values
```

Comment:
:x: Feels like we are lacking some terminology here. I wish we could say something like "The `=` operator can also _wire_ lazy values together", but I don't think we have defined that term before... 🤔